### PR TITLE
fix(cdk): improve type annotation for `tuiForAsync`

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/cdk/directives/for-async/for-async.directive.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export class TuiForAsyncDirective<T> implements OnChanges, OnDestroy {
     private readonly destroy$ = new Subject<void>();
 
     @Input()
-    tuiForAsyncOf: readonly T[] | null | undefined;
+    tuiForAsyncOf: T[] | readonly T[] | null | undefined;
```
Comment (nsbarsukov) [inline projects/cdk/directives/for-async/for-async.directive.ts:18]: It is strange behaviour 🤔 
`readonly T[]` should accepts `T[]`.

This example don't throws any error. Which typescript version you use ?
```ts
let a: ReadonlyArray<string> = [];
const b: string[] = ['1', '2'];

a = b;
```

Comment (splincode) [reply projects/cdk/directives/for-async/for-async.directive.ts:18]: Yes, but I don't understand why IDEA doesn't resolve it

Comment (waterplea) [reply projects/cdk/directives/for-async/for-async.directive.ts:18]: Maybe instead write `<T extends readonly any[]>` or something like that?

Comment (splincode) [reply projects/cdk/directives/for-async/for-async.directive.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Comment by splincode** [comment]
@waterplea @nsbarsukov @vladimirpotekhin Hello, sorry for ping, could you please review me?

**Comment by splincode** [comment]
@vladimirpotekhin could you review, please?

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**